docs/ql-documentation/learn-ql/javascript/flow-labels.rst

```diff
@@ -0,0 +1,391 @@
+<https://lgtm.com/projects/g/finos-plexus/plexus-interop/>`_ project on LGTM.com. Many of the 19
+results are false positives since we currently do not model many ways in which a value can be
+checked for nullness. In particular, after a property reference ``x.p`` we implicitly know that
+``x`` cannot be null anymore, since otherwise the reference would have thrown an exception.
+Modeling this would allow us to get rid of most of the false positives, but is beyond the scope of
+this tutorial.
+
+API
+---
+
+Plain data-flow configurations implicitly use a single flow label "data", which indicates that a
+data value originated from a source. You can use the predicate ``DataFlow::FlowLabel::data()``,
+which returns this flow label, as a symbolic name for it.
+
+Taint-tracking configurations add a second flow label "taint" (``DataFlow::FlowLabel::taint()``),
+which is similar to "data", but includes values that have passed through non-value preserving steps
+such as string operations.
+
+Each of the three member predicates ``isSource``, ``isSink`` and
+``isAdditionalFlowStep``/``isAdditionalTaintStep`` has one version that uses the default flow
+labels, and one version that allows specifying custom flow labels through additional arguments.
+
+For ``isSource``, there is one additional argument specifying which flow label(s) should be
+associated with values originating from this source. If multiple flow labels are specified, each
+value is associated with `all` of them.
+
+For ``isSink``, the additional argument specifies which flow label(s) a value that flows into this
+source may be associated with. If multiple flow labels are specified, then any value that is
+associated with `at least one` of them will be considered by the configuration.
+
+For ``isAdditionalFlowStep`` there are two additional arguments ``predlbl`` and ``succlbl``, which
+allow flow steps to act as flow label transformers. If a value associated with ``predlbl`` arrives
+at the start node of the additional step, it is propagated to the end node and associated with
+``succlbl``. Of course, ``predlbl`` and ``succlbl`` may be the same, indicating that the flow step
+preserves this label. There can also be multiple values of ``succlbl`` for a single ``predlbl`` or
+vice versa.
+
+Note that if you do not restrict ``succlbl`` then it will be allowed to range over all flow labels.
+This may cause labels that were previously blocked on a path to reappear, which is not usually what
+you want.
+
+The flow label-aware version of ``isBarrier`` is called ``isLabeledBarrier``: unlike ``isBarrier``,
+which prevents any flow past the given node, it only blocks flow of values associated with one of
+the specified flow labels.
+
+Standard queries using flow labels
+----------------------------------
+
+A few of our standard security queries already use flow labels. You can look at their implementation
```

Comment:
Why not future-proof this and write
```suggestion
Some of our standard security queries use flow labels. You can look at their implementation
```
I get that we intend to make more use of flow labels internally, but that should not be the concern of this tutorial.